keras/applications/nasnet.py

```diff
@@ -0,0 +1,649 @@
+            - `avg` means that global average pooling
+                will be applied to the output of the
+                last convolutional layer, and thus
+                the output of the model will be a
+                2D tensor.
+            - `max` means that global max pooling will
+                be applied.
+        classes: optional number of classes to classify images
+            into, only to be specified if `include_top` is True, and
+            if no `weights` argument is specified.
+        default_size: specifies the default image size of the model
+    # Returns
+        A Keras model instance.
+    # Raises
+        ValueError: in case of invalid argument for `weights`,
+            or invalid input shape.
+        RuntimeError: If attempting to run this model with a
+            backend that does not support separable convolutions.
+    '''
+    return NASNet(input_shape,
+                  penultimate_filters=1056,
+                  num_blocks=4,
+                  num_stem_block_filters=32,
+                  skip_reduction=False,
+                  filter_multiplier=2,
+                  include_top=include_top,
+                  weights=weights,
+                  input_tensor=input_tensor,
+                  pooling=pooling,
+                  classes=classes,
+                  default_size=224)
+
+
+def _separable_conv_block(ip, filters, kernel_size=(3, 3), strides=(1, 1),
+                          id=None):
+    '''Adds 2 blocks of [relu-separable conv-batchnorm]
+
+    # Arguments:
+        ip: input tensor
+        filters: number of output filters per layer
+        kernel_size: kernel size of separable convolutions
+        strides: strided convolution for downsampling
+        weight_decay: l2 regularization weight
+        id: string id
+
+    # Returns:
+        a Keras tensor
+    '''
+    channel_dim = 1 if K.image_data_format() == 'channels_first' else -1
+
+    with K.name_scope('separable_conv_block_%s' % id):
+        x = Activation('relu')(ip)
+        x = SeparableConv2D(filters, kernel_size, strides=strides,
+                            name='separable_conv_1_%s' % id,
+                            padding='same', use_bias=False,
+                            kernel_initializer='he_normal')(x)
+        x = BatchNormalization(axis=channel_dim, momentum=0.9997,
+                               epsilon=1e-3,
+                               name='separable_conv_1_bn_%s' % (id))(x)
+        x = Activation('relu')(x)
+        x = SeparableConv2D(filters, kernel_size,
+                            name='separable_conv_2_%s' % id,
+                            padding='same', use_bias=False,
+                            kernel_initializer='he_normal')(x)
+        x = BatchNormalization(axis=channel_dim, momentum=0.9997,
+                               epsilon=1e-3,
+                               name='separable_conv_2_bn_%s' % (id))(x)
+    return x
+
+
+def _adjust_block(p, ip, filters, id=None):
```

Comment:
Don't use `id`, it's a reserved keyword. Maybe `block_id` or something like that.